### ISSUE TITLE
Segfault in TableLoadSettings

### ISSUE THREAD

**Issue by sab24** [ocornut/imgui#3573]
Hi, I'm on 6c04124ce of the tables branch and my program fails to start with the only difference being a newer version of ImGui. Last version that worked was three weeks ago. As my MacOS based system is malfunctioning I only have a windows debugger output, and hope that I give enough data below.

Windows with gcc compiler using mingw
using imgui_impl_GLFW.cpp

```
    ImGui::BeginChild("##AthletesTable");

    int ncolumns = 9 + dataobj.readpoints.size(); //equals to 10 in this scenario
    ImVec2 size = ImVec2(2000, ImGui::GetTextLineHeightWithSpacing() * 30);

    static ImGuiTableFlags flags = ImGuiTableFlags_ScrollX | ImGuiTableFlags_ScrollY | ImGuiTableFlags_RowBg | ImGuiTableFlags_BordersOuter | ImGuiTableFlags_BordersV | ImGuiTableFlags_Resizable | ImGuiTableFlags_Reorderable | ImGuiTableFlags_Hideable;
    if (ImGui::BeginTable("##AthletesTable", ncolumns, flags, size)){
```

in imgui_widgets.cpp:10935 
```
    table->SettingsLoadedFlags = settings->SaveFlags;
``` 
the value of settings is a nullpointer.

stacktrace

```
>	T.exe!ImGui::TableLoadSettings(ImGuiTable * table) Line 10935	C++
 	T.exe!ImGui::BeginTableEx(volatile char * name, unsigned int id, int columns_count, int flags, volatile ImVec2 & outer_size, float inner_width) Line 8600	C++
 	T.exe!ImGui::BeginTable(volatile char * str_id, int columns_count, int flags, volatile ImVec2 & outer_size, float inner_width) Line 8403	C++
 	T.exe!MainWindow::showFirstTable(MainWindow * volatile this) Line 54	C++
```

```

  | Name | Value | Type
-- | -- | -- | --
◢ | settings | 0x0000000000000000 <NULL> | ImGuiTableSettings *
  | ID | <Unable to read memory> |  
  | SaveFlags | <Unable to read memory> |  
  | RefScale | <Unable to read memory> |  
  | ColumnsCount | <Unable to read memory> |  
  | ColumnsCountMax | <Unable to read memory> |  
  | WantApply | <Unable to read memory> |  
◢ | table | 0x0000000008e75e20 {ID=164868144 Flags=25185927 RawData={Size=1090 Capacity=1471 Data=0x0000000008e80290 "" } ...} | ImGuiTable *
  | ID | 164868144 | unsigned int
  | Flags | 25185927 | int
  | ▶ RawData | {Size=1090 Capacity=1471 Data=0x0000000008e80290 "" } | ImVector<char>
  | ▶ Columns | {Data=0x0000000008e80290 {ClipRect={Min={x=0.000000000 y=0.000000000 } Max={x=0.000000000 y=0.000000000 } } ...} ...} | ImSpan<ImGuiTableColumn>
  | ▶ DisplayOrderToIndex | {Data=0x0000000008e80678 "" DataEnd=0x0000000008e80682 "" } | ImSpan<signed char>
  | ▶ RowCellData | {Data=0x0000000008e80682 {BgColor=0 Column=0 '\0' } DataEnd=0x0000000008e806d2 {BgColor=0 Column=0 '\0' } } | ImSpan<ImGuiTableCellData>
  | VisibleMaskByIndex | 511 | unsigned __int64
  | VisibleMaskByDisplayOrder | 511 | unsigned __int64
  | VisibleUnclippedMaskByIndex | 3 | unsigned __int64
  | SettingsLoadedFlags | 0 | int
  | SettingsOffset | 4 | int
  | LastFrameActive | 25 | int
  | ColumnsCount | 10 | int
  | ColumnsVisibleCount | 9 | int
  | CurrentColumn | -1 | int
  | CurrentRow | -1 | int
  | InstanceCurrent | 0 | short
  | InstanceInteracted | -1 | short
  | RowPosY1 | 117.000000 | float
  | RowPosY2 | 117.000000 | float
  | RowMinHeight | 15.0000000 | float
  | RowTextBaseline | 0.000000000 | float
  | RowIndentOffsetX | 0.000000000 | float
  | RowFlags | 1 | int
  | LastRowFlags | 1 | int
  | RowBgColorCounter | 0 | int
  | ▶ RowBgColor | 0x0000000008e75ebc {4281544752} | unsigned int[1]
  | BorderColorStrong | 4284043087 | unsigned int
  | BorderColorLight | 4282399547 | unsigned int
  | BorderX1 | 68.0000000 | float
  | BorderX2 | 142.000000 | float
  | HostIndentX | 0.000000000 | float
  | OuterPaddingX | 1.00000000 | float
  | CellPaddingX | 4.00000000 | float
  | CellPaddingY | 2.00000000 | float
  | CellSpacingX1 | 1.00000000 | float
  | CellSpacingX2 | 0.000000000 | float
  | LastOuterHeight | 450.000000 | float
  | LastFirstRowHeight | 15.0000000 | float
  | InnerWidth | 0.000000000 | float
  | ColumnsTotalWidth | 175.000000 | float
  | ColumnsAutoFitWidth | 175.000000 | float
  | ResizedColumnNextWidth | 3.40282347e+38 | float
  | RefScale | 11.0000000 | float
  | ▶ OuterRect | {Min={x=68.0000000 y=117.000000 } Max={x=2068.00000 y=567.000000 } } | ImRect
  | ▶ WorkRect | {Min={x=68.0000000 y=117.000000 } Max={x=2054.00000 y=307370688. } } | ImRect
  | ▶ InnerClipRect | {Min={x=68.0000000 y=117.000000 } Max={x=142.000000 y=121.000000 } } | ImRect
  | ▶ BackgroundClipRect | {Min={x=68.0000000 y=117.000000 } Max={x=142.000000 y=121.000000 } } | ImRect
  | ▶ HostClipRect | {Min={x=68.0000000 y=117.000000 } Max={x=142.000000 y=121.000000 } } | ImRect
  | ▶ HostBackupParentWorkRect | {Min={x=68.0000000 y=117.000000 } Max={x=2054.00000 y=307370688. } } | ImRect
  | ▶ HostBackupClipRect | {Min={x=0.000000000 y=0.000000000 } Max={x=0.000000000 y=0.000000000 } } | ImRect
  | ▶ HostCursorMaxPos | {x=68.0000000 y=117.000000 } | ImVec2
  | ▶ HostBackupColumnsOffset | {x=0.000000000 } | ImVec1
  | ▶ OuterWindow | 0x0000000008e33ab0 {Name=0x0000000008e32fa0 "Hello, world22!/##AthletesTable_ADF3BAD2" ID=2465572148 ...} | ImGuiWindow *
  | ▶ InnerWindow | 0x0000000008e6a210 {Name=0x0000000008e6aa00 "Hello, world22!/##AthletesTable_ADF3BAD2/##AthletesTable_09D3B030" ...} | ImGuiWindow *
  | ▶ ColumnsNames | {Buf={Size=0 Capacity=72 Data=0x0000000008e6ab40 "Id" } } | ImGuiTextBuffer
  | ▶ DrawSplitter | {_Current=0 _Count=1 _Channels={Size=11 Capacity=15 Data=0x0000000008e2f550 {_CmdBuffer={Size=1 Capacity=...} ...} } } | ImDrawListSplitter
  | ▶ SortSpecsData | {Size=0 Capacity=0 Data=0x0000000000000000 <NULL> } | ImVector<ImGuiTableSortSpecsColumn>
  | ▶ SortSpecs | {Specs=0x0000000000000000 <NULL> SpecsCount=0 SpecsDirty=false ...} | ImGuiTableSortSpecs
  | SortSpecsCount | 0 '\0' | char
  | DeclColumnsCount | 0 '\0' | char
  | HoveredColumnBody | -1 'ÿ' | char
  | HoveredColumnBorder | -1 'ÿ' | char
  | ResizedColumn | -1 'ÿ' | char
  | LastResizedColumn | -1 'ÿ' | char
  | HeldHeaderColumn | -1 'ÿ' | char
  | ReorderColumn | -1 'ÿ' | char
  | ReorderColumnDir | 0 '\0' | char
  | RightMostVisibleColumn | -1 'ÿ' | char
  | LeftMostStretchedColumnDisplayOrder | -1 'ÿ' | char
  | ContextPopupColumn | -1 'ÿ' | char
  | FreezeRowsRequest | 0 '\0' | char
  | FreezeRowsCount | 0 '\0' | char
  | FreezeColumnsRequest | 0 '\0' | char
  | FreezeColumnsCount | 0 '\0' | char
  | RowCellDataCurrent | -1 'ÿ' | char
  | DummyDrawChannel | 10 '\n' | unsigned char
  | BgDrawChannelUnfrozen | 0 '\0' | unsigned char
  | IsLayoutLocked | false | bool
  | IsInsideRow | false | bool
  | IsInitializing | true | bool
  | IsSortSpecsDirty | true | bool
  | IsUsingHeaders | true | bool
  | IsContextPopupOpen | false | bool
  | IsSettingsRequestLoad | false | bool
  | IsSettingsDirty | false | bool
  | IsDefaultDisplayOrder | true | bool
  | IsResetDisplayOrderRequest | false | bool
  | IsUnfrozen | true | bool
  | MemoryCompacted | false | bool
  | HostSkipItems | false | bool
  | table->SettingsLoadedFlags | 0 | int

```

**Comment by ocornut** [comment]
Hello,
Thanks for the careful report.

`int ncolumns = 9 + dataobj.readpoints.size(); //equals to 10 in this scenario`

I managed to reproduce it with a varying column count (it only repro if you change that `ncolumns` value, because changing columns count discard old settings storage).

Looking at the fix now.

**Comment by ocornut** [comment]
Pushed a fix : 9039587afff25dbbdd9a7c8a096bb7cdc3c7ff35

**Comment by sab24** [comment]
Very fast! Really amazed by your productivity levels. Thanks for the help.
